router/core/src/main/scala/io/buoyant/router/Router.scala

```diff
@@ -255,11 +283,12 @@ object StackRouter {
      */
     def mkStack[Req, Rsp](orig: Stack[ServiceFactory[Req, Rsp]]): Stack[ServiceFactory[Req, Rsp]] =
       (orig ++ (TlsClientPrep.nop[Req, Rsp] +: stack.nilStack))
-        .replace(Retries.Role, XXX_ClassifierRequeueFilter.module[Req, Rsp])
   }
 
   def newPathStack[Req, Rsp]: Stack[ServiceFactory[Req, Rsp]] = {
     val stk = new StackBuilder[ServiceFactory[Req, Rsp]](stack.nilStack)
+    stk.push(StatsFilter.module)
+    stk.push(ClassifiedRetries.module)
```

Comment:
It's unclear to me if this is the proper ordering. it's consistent with finagle's client metrics, but I'm concerned that it will be difficult to compute a proper success rate with stats measured below retries.  I think for now we should move the StatsFilter above retries.  If we need stats below, we can add another module later.